plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,570 @@
+
+- name: Retrieve object metadata without object itself
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+    bucket_object_key: MyTestObjectKey
+
+- name: Retrieve detailed s3 object information
+  amazon.aws.s3_object_info:
+    bucket_name: MyTestBucket
+    bucket_object_key: MyTestObjectKey
+    object_details:
+      object_acl: true
+      object_tagging: true
+      object_legal_hold: true
+      object_attributes: true
+    object_attributes:
+      - ETag
+      - ObjectSize
+
+'''
+
+RETURN = r'''
+object_info:
+    description: S3 object details.
+    returned: always
+    type: complex
+    contains:
+        object_acl:
+            description: Access control list (ACL) of an object.
+            returned: when I(object_acl) is set to I(true).
+            type: complex
+            contains:
+                owner:
+                    description: Bucket owner's display ID and name.
+                    returned: always
+                    type: complex
+                    contains:
+                        id:
+                            description: Bucket owner's ID.
+                            returned: always
+                            type: str
+                            sample: "xxxxxxxxxxxxxxxxxxxxx"
+                        display_name:
+                            description: Bucket owner's display name.
+                            returned: always
+                            type: str
+                            sample: 'abcd'
+                grants:
+                    description: A list of grants.
+                    returned: always
+                    type: complex
+                    contains:
+                        grantee:
+                            description: The entity being granted permissions.
+                            returned: always
+                            type: complex
+                            contains:
+                                id:
+                                    description: The canonical user ID of the grantee.
+                                    returned: always
+                                    type: str
+                                    sample: "xxxxxxxxxxxxxxxxxxx"
+                                type:
+                                    description: type of grantee.
+                                    returned: always
+                                    type: str
+                                    sample: "CanonicalUser"
+                        permission:
+                            description: Specifies the permission given to the grantee.
+                            returned: always
+                            type: str
+                            sample: "FULL CONTROL"
+        object_legal_hold:
+            description: Object's current legal hold status
+            returned: when I(object_legal_hold) is set to I(true) and required configuration is set on bucket.
```

Comment:
Can you please explain what you mean with "configuration is set on bucket."?

Reply:
yes, so for example if user want to get object lock configuration info, `object_lock_configuration: true`, the `bucket` needs to have `Object Lock Configuration` set for bucket itself.
Otherwise `connection.get_object_legal_hold(**params)` will return boto error as 
```

ClientError('An error occurred (InvalidRequest) when calling the GetObjectLegalHold operation: 
Bucket is missing Object Lock Configuration')

```

One possible way to handle this instead I was thinking, is to just return an empty {} in such case where some configuration for the bucket does not exist, that way I can change documentation for that config to `returned: always`.

what do you think? @alinabuzachis @tremble